### PR TITLE
[REFACTORING] Slightly enhancing IMAP FETCH related code

### PR DESCRIPTION
**Pull request by chibenwa**
Non goals:
 - Kill `null` usages
 - Immutability everywhere and use `java.util.Stream`

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

File: protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/FetchResponseBuilder.java [hunk 1]
```diff
@@ -291,45 +289,37 @@ public void setInternalDate(Date internalDate) {
     }
 
     private FetchResponse.BodyElement wrapIfPartialFetch(Long firstOctet, Long numberOfOctets, FetchResponse.BodyElement fullResult) {
-        final FetchResponse.BodyElement result;
         if (firstOctet == null) {
-            result = fullResult;
+            return fullResult;
         } else {
```
Comment (rouazana) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/FetchResponseBuilder.java:283]: you could also remove most `else` in this commit

File: protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/EnvelopeBuilder.java [hunk 2]
```diff
@@ -63,13 +63,12 @@
...
         final Header header = MessageResultUtils.getMatching(headerName, message.headers());
-        final String result;
         if (header == null) {
-            result = null;
+            return null;
         } else {
             final String value = header.getValue();
             if (value == null || "".equals(value)) {
-                result = null;
+                return null;
```
Comment (rouazana) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/EnvelopeBuilder.java:71]: idem

File: protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/EnvelopeBuilder.java [hunk 3]
```diff
@@ -65,33 +65,28 @@ private String headerValue(Headers message, String headerName) throws MailboxExc
...
-                // IMAP-Servers are advised to also replace tabs with single spaces while doing the unfolding. This is what javamails
-                // unfold does. mime4j's unfold does strictly follow the rfc and so preserve them
-                //
-                // See IMAP-327 and https://mailman2.u.washington.edu/mailman/htdig/imap-protocol/2010-July/001271.html
-                return MimeUtility.unfold(value);
-
-            }
         }
+        final String value = header.getValue();
+        if (value == null || "".equals(value)) {
```
Comment (remk) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/EnvelopeBuilder.java:70]: not related to your change but line 110 we compare the trimmed value. Here not, shoud it be trimmed to ?

Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/EnvelopeBuilder.java:70]: (no idea)

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**